### PR TITLE
fix(server): override date via xmp

### PR DESCRIPTION
**Pull request by YFrendo**
The date have priority if it come from xmp
Related to #4502 and come from #5066

This ensures a date will be used from a sidecar file, if present. Prerequisite for #5066

### CODE REVIEW

File: server/src/domain/metadata/metadata.service.ts [hunk 1]
```diff
@@ -340,6 +340,31 @@ export class MetadataService {
...
+      // remove time from mediaTags, since sidecarTags has a more accurate time
+
+      mediaTags.SubSecDateTimeOriginal = undefined;
+      mediaTags.DateTimeOriginal = undefined;
+      mediaTags.SubSecCreateDate = undefined;
+      mediaTags.CreationDate = undefined;
+      mediaTags.CreateDate = undefined;
+      mediaTags.SubSecMediaCreateDate = undefined;
+      mediaTags.MediaCreateDate = undefined;
+      mediaTags.DateTimeCreated = undefined;
```
Comment (jrasm91) [inline server/src/domain/metadata/metadata.service.ts:365]: Settings things to undefined when they are being merged might lead to weird results in the future.

```typescript
const a = { CreateDate: 'placeholder' };
const b = { CreateDate: undefined };
const c = { ...a, ...b };
```
In this example c.CreateDate will actually be undefined, which I feel like is confusing and misleading. What do you think about just deleting from mediaTags list itself? So, `delete mediaTags.CreateDate` instead?

Comment (jrasm91) [reply server/src/domain/metadata/metadata.service.ts:365]: This is actually the same list of tags as above, so I think these need to be kept in sync. Maybe we can pass a list above, and then loop over this list here and delete the values.

Comment (YFrendo) [reply server/src/domain/metadata/metadata.service.ts:365]: > Settings things to undefined when they are being merged might lead to weird results in the future.
> 
> ```ts
> const a = { CreateDate: 'placeholder' };
> const b = { CreateDate: undefined };
> const c = { ...a, ...b };
> ```
> 
> In this example c.CreateDate will actually be undefined, which I feel like is confusing and misleading. What do you think about just deleting from mediaTags list itself? So, `delete mediaTags.CreateDate` instead?

look like a better idea indeed !

**Review by jrasm91 — APPROVED** [review]

**Comment by jjspierx** [comment]
How much work would it take for a PR that does this exact thing using the Google Photos JSON sidecars as well?  Support for these JSON sidecars would be very beneficial for those of using Google Takeout to export our photos.

**Pull request merged**